### ISSUE TITLE
Bug in `azurerm_synapse_spark_pool` when we set `dynamic_executor_allocation_enabled` to `true`

### ISSUE THREAD

**Issue by nooshin-mirzadeh** [hashicorp/terraform-provider-azurerm#17922]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.3

### AzureRM Provider Version

3.11.0

### Affected Resource(s)/Data Source(s)

azurerm_synapse_spark_pool

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "examplestorageacc"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  account_kind             = "StorageV2"
  is_hns_enabled           = "true"
}

resource "azurerm_storage_data_lake_gen2_filesystem" "example" {
  name               = "example"
  storage_account_id = azurerm_storage_account.example.id
}

resource "azurerm_synapse_workspace" "example" {
  name                                 = "example"
  resource_group_name                  = azurerm_resource_group.example.name
  location                             = azurerm_resource_group.example.location
  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
  sql_administrator_login              = "sqladminuser"
  sql_administrator_login_password     = "H@Sh1CoR3!"

  identity {
    type = "SystemAssigned"
  }
}

resource "azurerm_synapse_spark_pool" "example" {
  name                 = "example"
  synapse_workspace_id = azurerm_synapse_workspace.example.id
  node_size_family     = "MemoryOptimized"
  node_size            = "Small"
  cache_size           = 100
  
  dynamic_executor_allocation_enabled = true

  auto_scale {
    max_node_count = 50
    min_node_count = 3
  }
  
  auto_pause {
    delay_in_minutes = 15
  }

  tags = {
    ENV = "Production"
  }
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: updating Spark Pool: (Big Data Pool Name "small32" / Workspace Name "example" / Resource Group "example"): synapse.BigDataPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="ValidationFailed" Message="Spark pool request validation failed." Details=[{"code":"DynamicExecutorAllocationNotValid","message":"Min and max executors properties are mandatory fields when dynamic executor allocation is enabled"}]
```

### Expected Behaviour

Create the spark pool.

### Actual Behaviour

The error that it asks for the Min and max executors properties.

### Steps to Reproduce

If you run  `terraform apply` the code above, You will see the error when it reaches to applying the `azurerm_synapse_spark_pool`. 
It looks that recently, Azure has changed the configs, that when you set the `dynamic_executor_allocation_enabled` to `true`, it mandates you to define min and max executors, however, this is not implemented in the terraform provider and there is no property to set for min and max executor and the spark config won't work.
The problem exists only if we want to set the `dynamic_executor_allocation_enabled` to `true`.

**Comment by saithota321** [comment]
I'm facing the same issue. When is the issue going to be fixed?

**Comment by DataGobes** [comment]
This anomaly seems to be also present in the az synapse pool update command group for adding packages to a spark cluster with the azure CLI.

**Comment by roberthstrand** [comment]
I wanted to update this in the provider, but I see that the problem starts at the SDK level, where the deprecated package is used. I will not have the time to migrate it to the new package, it seems to be a lot of work getting it to run properly based on the migration instructions from Microsoft.

If anyone finds the time, please let me know if you need help testing.

**Comment by dlynch33** [comment]
Seems this has been addressed in version 3.29.0 of the Spark pool resource, with the addition of the min_executors and max_executors arguments:
https://registry.terraform.io/providers/hashicorp/azurerm/3.29.0/docs/resources/synapse_spark_pool
Great!

That update is not listed on the release notes however: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.29.0

**Comment by rcskosir** [comment]
@nooshin-mirzadeh Thanks for taking the time to submit this issue. It looks like this has been resolved as of v3.29.0. As such, I am going to mark this issue as closed. 

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new/choose) following the template. Thank you!
